pkg/endpoint/endpoint.go

```diff
@@ -2387,6 +2387,15 @@ func (e *Endpoint) syncPolicyMap() error {
 		e.realizedMapState = make(map[policymap.PolicyKey]struct{})
 	}
 
+	if e.desiredMapState == nil {
+		e.desiredMapState = make(map[policymap.PolicyKey]struct{})
+	}
+
+	if e.PolicyMap == nil {
```

Comment:
Should this be the first check we do? before the *MapState setups? It's harmless, I suppose, but I don't know if non-nil *MapState fields that are otherwise not filled in will cause confusion elsewhere.